tests/unit/test_discovery_client.py

```diff
@@ -91,6 +178,34 @@ def discovery_service_stub() -> FakeDiscoveryServiceStub:
     return FakeDiscoveryServiceStub()
 
 
+@pytest.fixture
+def discovery_service_temp_directory(tmp_path: pathlib.Path) -> pathlib.Path:
```

Comment:
Local fixtures don't have to have unique names. Consider renaming this to `temp_directory` for brevity.